### PR TITLE
feat: Add kripke package

### PR DESCRIPTION
**Pull request by wenkokke**

### CODE REVIEW

File: packages/control-kripke/src/Control/Kripke.hs [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+
+type (-->) :: (k -> Type) -> (k -> Type) -> (k -> Type)
+type (-->) f g x = f x -> g x
+
+data World :: Type
+
+type Path :: (World -> World -> Type) -> World -> World -> Type
+data Path step r t where
+  Refl :: Path step r r
+  Step :: Path step r s -> step s t -> Path step r t
```
Comment (cyberglot) [inline packages/control-kripke/src/Control/Kripke.hs]: `Step` should be `Trans` if you're using `Refl`.

File: packages/control-kripke/src/Control/Kripke.hs [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+  -- | Update a 'Kripke' value with a single 'step'.
+  updateWithStep :: step r t -> v r -> v t
+
+  -- | Update a 'Kripke' value with a 'Path'.
+  update :: Path step r t -> v r -> v t
+  update Refl v = v
+  update (Step prs st) v = updateWithStep st (update prs v)
+
+type Box :: (World -> World -> Type) -> (World -> Type) -> World -> Type
+type Box step v s = forall t. Path step s t -> v t
```
Comment (cyberglot) [inline packages/control-kripke/src/Control/Kripke.hs]: What is `Box`?

File: packages/control-kripke/src/Control/Kripke.hs [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+    m v s ->
+    (forall t. Path step s t -> Box step v t -> m w t) ->
+    m w s
+
+  tdrop ::
+    (Kripke step v, Kripke step w) =>
+    m v s ->
+    (forall t. Path step s t -> m w t) ->
+    m w s
+  tdrop mv mw = mv `tbind` \pst _ -> mw pst
```
Comment (cyberglot) [inline packages/control-kripke/src/Control/Kripke.hs]: Shouldn't they be `kpure`, `kbind` and `kdrop` ?

File: packages/control-kripke/src/Control/Kripke.hs [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+  where
+  Pure ::
+    forall step c v s.
+    v s ->
+    Free step c v s
+  Call ::
+    forall step c v w s.
+    c w s ->
+    All (Path step s --> w --> Free step c v) ->
+    Free step c v s
```
Comment (cyberglot) [inline packages/control-kripke/src/Control/Kripke.hs]: I'm not sure `Free` should be hanging out here with Kripke.

File: packages/control-kripke/src/Control/Kripke.hs (hunk 2 above)
Comment (wenkokke) [reply packages/control-kripke/src/Control/Kripke.hs]: Added comment.

File: packages/control-kripke/src/Control/Kripke.hs (hunk 1 above)
Comment (wenkokke) [reply packages/control-kripke/src/Control/Kripke.hs]: No, this is a normalised reflexive transitive closure, but I'll rename `compose` to `trans`.

File: packages/control-kripke/src/Control/Kripke.hs (hunk 4 above)
Comment (wenkokke) [reply packages/control-kripke/src/Control/Kripke.hs]: It's the free *Kripke* monad, so I think it should be here.

**Pull request merged**